packages/components/src/internal/components/samples/SampleSetCards.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { FC, memo, useEffect, useMemo } from 'react';
+
+import { Filter } from '@labkey/api';
+
+import { Alert, Cards, caseInsensitive, naturalSort, LoadingSpinner, SCHEMAS } from '../../..';
+
+import { ICardProps } from '../base/Cards';
+
+// These need to be direct imports from files to avoid circular dependencies in index.ts
+import { InjectedQueryModels, QueryConfigMap, withQueryModels } from '../../../public/QueryModel/withQueryModels';
+
+const getSampleSetCount = (row: Record<string, any>): number => {
+    return caseInsensitive(row, 'SampleCount').value;
+};
+
+const getSampleSetDescription = (row: Record<string, any>): string => {
+    return caseInsensitive(row, 'Description').value;
+};
+
+const getSampleSetName = (row: Record<string, any>): string => {
+    return caseInsensitive(row, 'Name').value;
+};
+
+const rowSort = (row1: Record<string, any>, row2: Record<string, any>): number => {
+    return naturalSort(getSampleSetName(row1), getSampleSetName(row2));
+};
+
+const getSampleSetCard = (row: Record<string, any>): ICardProps => {
+    const sampleCount = getSampleSetCount(row);
+    const noSamples = sampleCount === 0;
+
+    return {
+        caption: noSamples ? getSampleSetDescription(row) : 'Sample count: ' + sampleCount,
+        disabled: noSamples,
+        href: caseInsensitive(row, 'Name').url,
+        iconSrc: noSamples ? 'sample_set_gray' : 'sample_set',
+        title: getSampleSetName(row),
+    };
+};
+
+const getNonEmptySampleSetCards = (data: Array<{ [key: string]: any }>): ICardProps[] => {
+    return data
+        .filter(row => getSampleSetCount(row) > 0)
+        .sort(rowSort)
+        .map(getSampleSetCard);
+};
+
+const getEmptySampleSetCards = (data: Array<{ [key: string]: any }>): ICardProps[] => {
+    return data
+        .filter(row => getSampleSetCount(row) === 0)
+        .sort(rowSort)
+        .map(getSampleSetCard);
+};
+
+interface Props {
+    modelId: string;
+}
+
+const SampleSetCardsImpl: FC<Props & InjectedQueryModels> = memo(({ actions, modelId, queryModels }) => {
+    const model = queryModels[modelId];
+
+    useEffect(() => {
```

Comment:
nit: No need for this `useEffect` if `autoLoad` is set to `true`.

Reply:
Ah yes, did that before I discovered autoload